### ISSUE TITLE
Finetuning Multilingual Model

### ISSUE THREAD

**Issue by nikhiljaiswal** [pytorch/fairseq#3343]
Hi,

I want to finetune the m2m model on my dataset which contains en and de in the source and its corresponding de and en in the target language. In other words, I want to perform joint training of en-de and de-en. I have searched the docs but unable to find any such solution. How to prepare the binarize data since we have both en & de in the source corpus, what parameters we have to pass?

**Comment by jaspock** [comment]
This [answer](https://github.com/pytorch/fairseq/issues/3233#issuecomment-791579699)  may help you.

**Comment by nikhiljaiswal** [comment]
Hi @jaspock , thanks for the response. I went through that answer and tried with transformer_wmt_en_de_big, still I am getting the error as architecture do not match. Please help.

I tried using following - 

fairseq-train $path_2_data
--finetune-from-model $pretrained_model
--max-epoch 500
--ddp-backend=legacy_ddp
--task translation_multi_simple_epoch
--lang-pairs de-en,en-de
--arch transformer_wmt_en_de_big
--share-decoder-input-output-embed
--optimizer adam --adam-betas '(0.9, 0.98)'
--lr 0.0005 --lr-scheduler inverse_sqrt
--warmup-updates 4000 --warmup-init-lr '1e-07'
--label-smoothing 0.1 --criterion label_smoothed_cross_entropy
--dropout 0.3 --weight-decay 0.0001
--max-tokens 4000
--update-freq 8

**Comment by jaspock** [comment]
See my [new answer](https://github.com/pytorch/fairseq/issues/3233#issuecomment-802020438).
